'import unyt' fails: ImportError: cannot import name '__version__' from 'unyt._version'

* unyt version: 3.0.1
* Python version: 3.9
* Operating System: FreeBSD 13.2

### Description

In version 3.0.1 'import unyt' fails, and tests fail:
```
____________________________________________________________________ ERROR collecting unyt/tests/test_array_functions.py ____________________________________________________________________
ImportError while importing test module '/usr/ports/math/py-unyt/work-py39/unyt-3.0.1/unyt/tests/test_array_functions.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../__init__.py:62: in <module>
    from ._version import __version__
E   ImportError: cannot import name '__version__' from 'unyt._version' (/usr/ports/math/py-unyt/work-py39/unyt-3.0.1/unyt/_version.py)
______________________________________________________________________ ERROR collecting unyt/tests/test_dask_arrays.py ______________________________________________________________________
ImportError while importing test module '/usr/ports/math/py-unyt/work-py39/unyt-3.0.1/unyt/tests/test_dask_arrays.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../__init__.py:62: in <module>
    from ._version import __version__
E   ImportError: cannot import name '__version__' from 'unyt._version' (/usr/ports/math/py-unyt/work-py39/unyt-3.0.1/unyt/_version.py)
```

Tests ran fine in 2.9.5

### What I Did

pytest

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [build-system]
 requires = [
   "setuptools>=61.2",
-  "setuptools_scm[toml]>=6.2",
+  "setuptools_scm[toml]>=7.0.1",
 ]
 
 build-backend = "setuptools.build_meta"

